js/components/markers/markerIcon.jsx

```diff
@@ -95,21 +95,22 @@ class MarkerIcon extends Component {
    */
   getContent() {
     let content;
-    const { data } = this.props;
+    const { data, config } = this.props;
     const { hover } = this.state;
+    const marker = Object.assign({}, config, data);
 
-    switch (data.type) {
+    switch (marker.type) {
       case CONSTANTS.MARKERS.TYPE.TEXT: {
-        if (!data.text || data.text.length === 0) {
+        if (!marker.text || marker.text.length === 0) {
           return null;
         }
-        const truncatedText = data.text.length <= CONSTANTS.MARKERS.MAXCHAR
-          ? data.text : data.text.slice(0, CONSTANTS.MARKERS.MAXCHAR).concat(' ...');
+        const truncatedText = marker.text.length <= CONSTANTS.MARKERS.MAXCHAR
+          ? marker.text : marker.text.slice(0, CONSTANTS.MARKERS.MAXCHAR).concat(' ...');
         content = <p>{truncatedText}</p>;
```

Comment:
Isn't it better to return immediately as you do on lines 105 and 114 and avoid using an additional variable?

Reply:
I can do it, but I will need to disable the following rule on the ESLINT config:
https://eslint.org/docs/rules/no-unreachable

Reply:
Nervermind, you are right....since I'm not doing anything else I can avoid the usage of the `break` statement and simply return the desired content...thank you for the observation!!!